cmd/csi_driver/main.go

```diff
@@ -42,13 +55,203 @@ var (
 	version = "unknown"
 )
 
+func getGvnicNames() ([]string, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list network interfaces: %w", err)
+	}
+
+	var ethNics []string
+	for _, link := range links {
+		if strings.HasPrefix(link.Attrs().Name, "eth") {
+			ethNics = append(ethNics, link.Attrs().Name)
+		}
+	}
+
+	return ethNics, nil
+}
+
+func configureRoutesForNic(nicName string, ipAddr net.IP, tableID int) error {
+	// Get the network interface handle
+	link, err := netlink.LinkByName(nicName)
+	if err != nil {
+		return fmt.Errorf("failed to get interface %s: %w", nicName, err)
+	}
+
+	// Find the gateway for this interface
+	routes, err := netlink.RouteList(link, netlink.FAMILY_V4)
+	if err != nil {
+		return fmt.Errorf("failed to list routes for %s: %w", nicName, err)
+	}
+
+	var gateway net.IP
+	for _, r := range routes {
+		if r.Gw != nil {
+			gateway = r.Gw
+
+			break
+		}
+	}
+
+	if gateway == nil {
+		return fmt.Errorf("could not find gateway for %s", nicName)
+	}
+	klog.Infof("Found gateway %s for %s", gateway.String(), nicName)
+
+	// Define and add the route
+	_, dst, _ := net.ParseCIDR("10.170.0.0/16") // MGS IP
```

Comment:
Noted this is the IP of the lustre instance which is passed in through volumeContext